### PR TITLE
http2: Flip the upstream H2 frame flood and abuse checks to ON by default

### LINKED ISSUE #12281: [HTTP/2] extend flood protection to upstream servers
Flood protection checks are currently enabled in downstream (server) codecs. These checks need to be extended to the upstream client codecs as well to allow Envoy to work with untrusted upstream servers.

Depends on #12280
Sub-task of #12278

### PR DESCRIPTION
**Pull request by yanavlasov**
Additional Description:
This PR switches the upstream flood and abuse checks to on by default. It can be disabled through a runtime flag.

Risk Level: High (upstream H2 codec behavior)
Testing: Unit, Integration tests
Docs Changes: N/A
Release Notes: Yes
Platform Specific Features: N/A
Runtime guard: envoy.reloadable_features.upstream_http2_flood_checks
Fixes #12281

Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

**Review by alyssawilk — COMMENTED** [review]
I'm really glad we're at the point we're ready to flip this, but if we consider it high risk I wonder if we want to delay?
AFIK most companies are in their holiday freeze right, the envoy-project is going "on vacation" for two weeks, then we come back and immediately cut the release without it getting production burn time. On the flip side, it is runtime guarded and we do point releases now, so maybe I'm erring too far on the side of caution.
cc @envoyproxy/senior-maintainers WDYT?

**Comment by ggreenway** [comment]
I agree with not wanting to merge high-risk right now. But another factor is that the 1.17 release is scheduled for 12/31, so if we want this to be included in 1.17, it should probably go in now instead of right before we cut the release (which I'm sure won't actually happen on NYE; first or second week of january is likely).

**Comment by alyssawilk** [comment]
Yeah, 1.17 is my point - if we cut the release in the first week (which per
our release guidance we're supposed to try harder to do than we used to =P)
I think it wouldn't get burn time in prod.
So do we want it in (high risk, but runtime flippable) or wait until after
the release?

**Comment by mattklein123** [comment]
I would vote for holding this until after the release, but I don't feel strongly about it if there is a strong interest in merging now.

**Comment by ggreenway** [comment]
I'd say wait until after the release. Even if it went in now, but I bet very few people are about to run master in production over a long break, so it won't have much burn-in time before the release.

File: source/common/http/http2/protocol_constraints.cc
Comment (mattklein123) [inline source/common/http/http2/protocol_constraints.cc:13]: I'm confused. Why are these changes needed? Or was this a previous bug and unrelated to the flip?

/wait-any

Comment (yanavlasov) [reply source/common/http/http2/protocol_constraints.cc]: Good catch. These were left over work in progress. It just needed to fix CodecStreamIdleTimeout so that it provides fully initialized http2 options.

**Review by mattklein123 — APPROVED** [review]

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**
